File: src/unity/Runtime/IronSource/IIronSource.cs
using System.Threading.Tasks;

namespace EE {
    public interface IIronSource : IPlugin {
        Task<bool> Initialize(string appKey);
        IBannerAd CreateBannerAd(string adId, IronSourceBannerAdSize adSize);
        IFullScreenAd CreateInterstitialAd(string adId);
        IFullScreenAd CreateRewardedAd(string adId);
    }
}

File: src/ts/src/iron_source/IIronSource.ts
import {
    IBannerAd,
    IFullScreenAd,
} from "../ads";
import { IPlugin } from "../core";
import { IronSourceBannerAdSize } from "./IronSourceBannerAdSize";

export interface IIronSource extends IPlugin {
    initialize(appKey: string): Promise<boolean>;
    createBannerAd(adId: string, adSize: IronSourceBannerAdSize): IBannerAd;
    createInterstitialAd(adId: string): IFullScreenAd;
    createRewardedAd(adId: string): IFullScreenAd;
}